Towny/src/main/java/com/palmergames/bukkit/towny/command/PlotCommand.java

```diff
@@ -1547,6 +1555,84 @@ public void parsePlotGroupAdd(String[] split, TownBlock townBlock, Player player
 			TownyMessaging.sendMsg(player, Translatable.of("msg_plot_was_put_into_group_x", townBlock.getX(), townBlock.getZ(), townBlock.getPlotObjectGroup().getName()));
 		}
 	}
+	
+	public void parsePlotGroupFill(String[] split, TownBlock townBlock, Player player, Town town, boolean isDelete) throws TownyException {
+		if (isDelete) 	checkPermOrThrow(player, PermissionNodes.TOWNY_COMMAND_PLOT_GROUP_ADD.getNode());
+		else 			checkPermOrThrow(player, PermissionNodes.TOWNY_COMMAND_PLOT_GROUP_REMOVE.getNode());
+
+		Resident resident = getResidentOrThrow(player);
+
+		if (split.length != 2 && !resident.hasPlotGroupName())
+			throw new TownyException(Translatable.of("msg_err_plot_group_name_required"));
+
+		WorldCoord playerWorldCoord = WorldCoord.parseWorldCoord(player);
+		
+		//сюда записать все блоки города заполненные по алгоритму floodfill
+		List<TownBlock> townBlockList = new ArrayList<>();
+
+		final Set<WorldCoord> valid = new HashSet<>();
+		final Set<WorldCoord> visited = new HashSet<>();
+
+		final Queue<WorldCoord> queue = new LinkedList<>();
+		
+		
+		queue.offer(playerWorldCoord);
+		visited.add(playerWorldCoord);
+		
+		final int[][] DIRECTIONS = {{-1, 0}, {1, 0}, {0, -1}, {0, 1}};
+		while (!queue.isEmpty()) {
+			final WorldCoord current = queue.poll();
+
+			valid.add(current);
+
+			for (final int[] direction : DIRECTIONS) {
+				final int xOffset = direction[0];
+				final int zOffset = direction[1];
+				
+				//in this code not needed check Max and min because town is not infinity
+
+				final WorldCoord candidate = current.add(xOffset, zOffset);
+
+				final TownBlock checked_townBlock = candidate.getTownBlockOrNull();
+				
+					if (checked_townBlock != null &&
+						checked_townBlock.hasTown() &&
+						checked_townBlock.getTown().equals(town) &&
+						checked_townBlock.getType().equals(townBlock.getType()) &&
+						!visited.contains(candidate)) {
+
+						queue.offer(candidate);
+						visited.add(candidate);
+					}
+			}
+		}
+		
+		for (WorldCoord worldCoord : valid) {
+			TownBlock townBlock1 = worldCoord.getTownBlockOrNull();
+			
+			if (isDelete) {
+				if (townBlock != null) {
+					try {
+						parsePlotGroupRemove(townBlock1, player, town);
+
+					} catch (TownyException e) {
+						//I don't know, for now I'll do it so that I can go through each chunk from the set
+					}
+				}
+			}
+			else {
+				if (townBlock1 != null) {
+					try {
+						parsePlotGroupAdd(split, townBlock1, player, town);

```

Comment:
If your code has picked up any plotgroups that aren't the one involved in the fill, which are different plotgroups, the player is going to be spammed with Confirmations about transferring each plot into the filled plotgroup.

The `valid` set should not be able to receive plots which are already part of a different plotgroup.